### PR TITLE
Zeta zero on boundary

### PR DESCRIPTION
**Pull request by matt-frey**

### CODE REVIEW

File: src/3d/utils/parameters.f90 [hunk 1]
```diff
@@ -114,4 +127,40 @@ subroutine update_parameters
...
+    subroutine set_zeta_boundary_flag(zeta, thres)
+        double precision, intent(in) :: zeta(-1:nz+1, 0:ny-1, 0:nx-1)
+        double precision, intent(in) :: thres
+        double precision :: rms_bndry(2), rms_interior
+
+        rms_interior = dsqrt(sum(zeta(1:nz-1, :, :) ** 2) * nhcelli / (nz-1))
+        rms_bndry(1) = dsqrt(sum(zeta(0,      :, :) ** 2) * nhcelli)
+        rms_bndry(2) = dsqrt(sum(zeta(nz,     :, :) ** 2) * nhcelli)
+
+        l_bndry_zeta_zero(:) = (rms_bndry(:) < thres * (epsilon(rms_interior) + rms_interior))
```
Comment (sjboeing) [inline src/3d/utils/parameters.f90:141]: Is "thres*rms_interior+epsilon(rms_interior)" better?

Comment (matt-frey) [reply src/3d/utils/parameters.f90:141]: Yes, we can also use this criterion. Maybe better as with the other eq. you need a large thres value.

File: src/3d/utils/parameters.f90 [hunk 2]
```diff
@@ -114,4 +130,81 @@ subroutine update_parameters
...
+
+
+    subroutine write_zeta_boundary_flag(ncid)
+        integer, intent(in)     :: ncid
+        integer                 :: grp_ncid
+        character(*), parameter :: name = 'parameters'
+
+        ncerr = nf90_def_grp(ncid, name, grp_ncid)
+
+        call check_netcdf_error("Faild to create NetCDF group '" // name // "'.")
```
Comment (sjboeing) [inline src/3d/utils/parameters.f90:186]: Failed

Comment (matt-frey) [reply src/3d/utils/parameters.f90:186]: Thanks ... I make many English typos.

**Comment by matt-frey** [comment]
I thought we justified our approach. @daviddritschel What do you think?

**Comment by sjboeing** [comment]
(if it needs changing, this could still be done in a later PR in any case)

**Comment by matt-frey** [comment]
I prefer having a final good solution now. We can add an option if necessary that forces zero on the boundary.

**Pull request merged**